cmd/soroban-cli/src/config/mod.rs

```diff
@@ -1,114 +1,102 @@
 use std::path::PathBuf;
 
 use clap::{arg, command};
+use secret::StellarSigner;
 use serde::{Deserialize, Serialize};
 
-use soroban_rpc::Client;
+use crate::signer::{self, Stellar};
+use crate::xdr::{Transaction, TransactionEnvelope};
+use crate::Pwd;
 
-use crate::{
-    signer,
-    xdr::{Transaction, TransactionEnvelope},
-    Pwd,
-};
-
-use self::{network::Network, secret::Secret};
+use self::network::Network;
 
 pub mod alias;
 pub mod data;
 pub mod locator;
 pub mod network;
 pub mod secret;
+pub mod sign_with;
 
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
     #[error(transparent)]
-    Network(#[from] network::Error),
+    SignWith(#[from] sign_with::Error),
     #[error(transparent)]
     Secret(#[from] secret::Error),
     #[error(transparent)]
-    Config(#[from] locator::Error),
+    Network(#[from] network::Error),
     #[error(transparent)]
-    Rpc(#[from] soroban_rpc::Error),
+    Locator(#[from] locator::Error),
     #[error(transparent)]
     Signer(#[from] signer::Error),
 }
 
 #[derive(Debug, clap::Args, Clone, Default)]
 #[group(skip)]
 pub struct Args {
-    #[command(flatten)]
-    pub network: network::Args,
-
     #[arg(long, visible_alias = "source", env = "STELLAR_ACCOUNT")]
     /// Account that signs the final transaction. Alias `source`. Can be an identity (--source alice), a secret key (--source SC36…), or a seed phrase (--source "kite urban…").
     pub source_account: String,
 
-    #[arg(long)]
-    /// If using a seed phrase, which hierarchical deterministic path to use, e.g. `m/44'/148'/{hd_path}`. Example: `--hd-path 1`. Default: `0`
-    pub hd_path: Option<usize>,
-
     #[command(flatten)]
-    pub locator: locator::Args,
+    pub sign_with: sign_with::Args,
```

Comment:
I'm not convinced that we want to put locator inside `sign_with`. Isn't it a separate logical entity? (As far as flags go)?
We can pass locator into `.signer()` command, as we need it to locate the private key

Reply:
Soon we are going to refactor this out completely since now it is treated as global and we can have the root command pass it. So we will need to update the interface to pass the locator around. However, the locator is flattened into this and then flattened into the general config::Args. So I don't see any harm in it for now and we can refactor it away in a future PR.